help_function/src/util_group_store_test.cpp

```diff
@@ -0,0 +1,270 @@
+    size_t temp_storage_size = group_store::get_local_memory_size(128);
+    sycl::local_accessor<uint8_t, 1> tacc(sycl::range<1>(temp_storage_size), h);
+    sycl::accessor dacc_read(buffer, h, sycl::read_only);
+    sycl::accessor dacc_write(buffer_out, h, sycl::read_write);
+    
+    h.parallel_for(
+        sycl::nd_range<3>(sycl::range<3>(1, 1, 128), sycl::range<3>(1, 1, 128)),
+        [=](sycl::nd_item<3> item) {
+          int thread_data[4];
+          int global_index =
+              item.get_group(2) * item.get_local_range().get(2) +
+              item.get_local_id(2); 
+          #pragma unroll
+          for (int i = 0; i < 4; ++i) {
+            thread_data[i] = dacc_read[global_index * 4 + i];
+          }
+          
+          auto *d_w =
+              dacc_write.get_multi_ptr<sycl::access::decorated::yes>().get();
+          auto *tmp = tacc.get_multi_ptr<sycl::access::decorated::yes>().get();
+          
+          // Store thread_data of each work item from blocked arrangement
+          group_store(tmp).store(item, d_w, thread_data);
+          
+          });
+  });
+  q.wait_and_throw();
+
+  sycl::host_accessor data_accessor(buffer_out, sycl::read_write);
+  const int *ptr = data_accessor.get_multi_ptr<sycl::access::decorated::yes>();
+  return helper_validation_function<S>(ptr, "test_group_store");
+}
+
+
+bool test_store_subgroup_striped_standalone() {
+  // Tests dpct::group::load_subgroup_striped as standalone method
+  sycl::queue q(dpct::get_default_queue());
+  int data[512];
+  for(int i=0;i<512;i++){data[i]=i;}
+  sycl::buffer<int, 1> buffer(data, 512);
+  sycl::buffer<uint32_t, 1> sg_sz_buf{sycl::range<1>(1)};
+  int data_out[512];
+  sycl::buffer<int, 1> buffer_out(data_out, 512);
+
+  q.submit([&](sycl::handler &h) {
+    sycl::accessor dacc_read(buffer, h, sycl::read_only);
+    sycl::accessor dacc_write(buffer_out, h, sycl::read_write);
+    sycl::accessor sg_sz_dacc(sg_sz_buf, h, sycl::read_write);
+    
+    h.parallel_for(
+        sycl::nd_range<3>(sycl::range<3>(1, 1, 128), sycl::range<3>(1, 1, 128)),
+        [=](sycl::nd_item<3> item) {
+          int thread_data[4];
+          int global_index =
+              item.get_group(2) * item.get_local_range().get(2) +
+              item.get_local_id(2); 
+          #pragma unroll
+          for (int i = 0; i < 4; ++i) {
+            thread_data[i] = dacc_read[global_index * 4 + i];
+          }
+          auto *d_w =
+              dacc_write.get_multi_ptr<sycl::access::decorated::yes>().get();
+          auto *sg_sz_acc =
+              sg_sz_dacc.get_multi_ptr<sycl::access::decorated::yes>().get();
+          size_t gid = item.get_global_linear_id();
+          if (gid == 0) {
+            sg_sz_acc[0] = item.get_sub_group().get_local_linear_range();
+          }
+          dpct::group::store_subgroup_striped<4, int>(item, d_w, thread_data);
+          // reapply global mapping

```

Comment:
We are overwriting the result of the sub-group store here from lines 173 to 179. If we remove this the test fails.

Reply:
Interesting... seems like this should not be here.